src/scrap.py

```diff
@@ -0,0 +1,262 @@
+
+    reddit_jobs = []
+
+    for element in div_post:
+        header_title = element.find("h3", class_="_eYtD2XCVieq6emjKBH3m")
+        link_job = element.find("a", class_="_3t5uN8xUmg0TOwRCOGQEcU")
+
+        job = {}
+
+        if link_job is not None:
+            job["_title"] = header_title.text
+            job["_link"] = link_job.get("href")
+
+            reddit_jobs.append(job)
+
+    return reddit_jobs
+
+
+# -------------------FOR RAPIDAPI----------------------------
+
+
+def rapid_response() -> List[object]:
+    url = "https://indeed12.p.rapidapi.com/jobs/search?locality=us&query=software+engineer&location=Fullerton&formatted_relative_time=today&page_id=1"
+
+    rapid_api_key = os.getenv("RAPID_API_KEY")
+    rapid_api_host = os.getenv("RAPID_API_HOST")
+
+    headers = {
+        "X-RapidAPI-Key": rapid_api_key,
+        "X-RapidAPI-Host": rapid_api_host,
+    }
+
+    rapid_jobs = []
+    response = requests.get(url, headers=headers).json()
+
+    for elem in response["hits"]:
+        job = {}
+        job["_company"] = elem["company_name"]
+        job["_title"] = elem["title"]
+        job["_location"] = elem["location"]
+        job["_link"] = f'https://www.indeed.com/viewjob?jk={elem["id"]}&locality=us'
+
+        rapid_jobs.append(job)
+
+    return rapid_jobs
+
+
+# -------------------FOR INDEED API----------------------------
+
+
+# -------------------FOR LINKEDIN----------------------------
+
+
+def indeed_response() -> List[object]:
+    url = os.getenv("LINKEDIN_URL")
+
+    response = requests.get(url)
+
+    content = response.text
+
+    parse_content = BeautifulSoup(content, "html.parser")
+
+    div_post = parse_content.find_all(
+        "div",
+        class_="base-card relative w-full hover:no-underline focus:no-underline base-card--link base-search-card base-search-card--link job-search-card",
+    )
+
+    indeed_jobs = []
+
+    for elem in div_post:
+        job = {}
+        job["_company"] = elem.find(
+            "h4", class_="base-search-card__subtitle"
+        ).text.strip()
+        job["_title"] = elem.find("h3", class_="base-search-card__title").text.strip()
+        job["_location"] = elem.find(
+            "span", class_="job-search-card__location"
+        ).text.strip()
+        job["_link"] = elem.find("a", class_="base-card__full-link")["href"]
+
+        indeed_jobs.append(job)
+
+    return indeed_jobs
+
+
+# -------------------FOR HELPER FUNCTIONS (possible in class)----------------------------
+
+# List opportunities shows all the opportunities that have NOT been posted yet
+
+
+def list_opportunities() -> List[object]:
+    connection = sqlite3.connect("jobs.db")
+    cursor = connection.cursor()
+
+    cursor.execute("SELECT * FROM jobs")
+    rows = cursor.fetchall()
+
+    # Printing out for debugging purposes
+    for row in rows:
+        _company = row[0]
+        _title = row[1]
+        _location = row[2]
+        _link = row[3]
+        _processed = row[4]
```

Comment:
can be one linered w `_company, _title, _location, _link, _processed = zip(*row)`